### PR TITLE
No change mdb (Exomol, HITRAN, Hitemp) from opa 

### PR DESCRIPTION
**Pull request by HajimeKawahara**
address the side effect raised in #510 

The following is a simple test of the side effect:

`tests/unittest/opa/sideeffect_test.py`
```python
def test_sideeffect_call():
    nus, wav, res = wavenumber_grid(
        22920.0, 23100.0, 20000, unit="AA", xsmode="premodit"
    )
    mdb = mock_mdbHitemp(multi_isotope=True)
    opa1 = OpaPremodit(mdb, nu_grid=nus, allow_32bit=True, auto_trange=[500.0, 1000])
    opa1_orig = copy.deepcopy(opa1)
    opa2 = OpaPremodit(
        mdb, nu_grid=nus, allow_32bit=True, auto_trange=[500.0, 1200]
    )  # used the same mdb used in opa1
    print(opa1 == opa1_orig) -> True 
``` 
Before this PR (i.e. recent `develop` branch), `opa1 == opa1_orig` gives `False`. This means the second call of `OpaPremodit` using the same `mdb` changes `opa1.mdb`. In this PR, based on the strategy described in #510, we avoid the side effect, i.e. `opa` does not change the internal state of `mdb`. 

but, we need some modifications:
- removed the internal state `mdb.Tref`
- `line_strength_ref_original` was introduced for line strength at `T=Tref_original`
- `self.line_strength_ref` (which depends on `self.Tref`) -> `self.line_strength(Tref)` 
- `qr_interp(T)` -> `qr_interp(T,Tref)` in `MdbExomol`
- `qr_interp(iso,T)` -> `qr_interp(iso,T,Tref)` in `MdbHitemp/Hitran`
- `qr_interp_line(T)` -> `qr_interp_line(T,Tref)` in `MdbHitemp/Hitran`

- The deprecated attribute `Sij0` was removed.

### CODE REVIEW

**Review by ykawashima — APPROVED** [review]
Thank you for the modification. I have confirmed that the edit yields the desirable result for "print(opa1 == opa1_orig)" in the sample code.

File: src/exojax/spec/opacalc.py [hunk 1]
```diff
@@ -333,9 +333,9 @@ def xsvector(self, T, P):
         nsigmaD = normalized_doppler_sigma(T, self.mdb.molmass, R)
 
         if self.mdb.dbtype == "hitran":
-            qt = self.mdb.qr_interp(self.mdb.isotope, T)
+            qt = self.mdb.qr_interp(self.mdb.isotope, T, Tref_original)
         elif self.mdb.dbtype == "exomol":
-            qt = self.mdb.qr_interp(T)
+            qt = self.mdb.qr_interp(T, Tref_original)
```
Comment (HajimeKawahara) [inline src/exojax/spec/opacalc.py:338]: `qt = self.mdb.qr_interp(T, Tref_original)` should be `qt = self.mdb.qr_interp(T, self.Tref_broadening)`

File: src/exojax/spec/opacalc.py [hunk 2]
```diff
@@ -333,9 +333,9 @@ def xsvector(self, T, P):
         nsigmaD = normalized_doppler_sigma(T, self.mdb.molmass, R)
 
         if self.mdb.dbtype == "hitran":
-            qt = self.mdb.qr_interp(self.mdb.isotope, T)
+            qt = self.mdb.qr_interp(self.mdb.isotope, T, Tref_original)
```
Comment (HajimeKawahara) [inline src/exojax/spec/opacalc.py:336]: `qt = self.mdb.qr_interp(self.mdb.isotope, T, Tref_original)` should be `qt = self.mdb.qr_interp(self.mdb.isotope, T, self.Tref_broadening)`

File: src/exojax/spec/opacalc.py [hunk 3]
```diff
@@ -420,9 +420,9 @@ def xsmatrix(self, Tarr, Parr):
         ) = self.opainfo
 
         if self.mdb.dbtype == "hitran":
-            qtarr = vmap(self.mdb.qr_interp, (None, 0))(self.mdb.isotope, Tarr)
+            qtarr = vmap(self.mdb.qr_interp, (None, 0, None))(self.mdb.isotope, Tarr, Tref_original)
         elif self.mdb.dbtype == "exomol":
-            qtarr = vmap(self.mdb.qr_interp)(Tarr)
+            qtarr = vmap(self.mdb.qr_interp, (0, None))(Tarr, Tref_original)
```
Comment (HajimeKawahara) [inline src/exojax/spec/opacalc.py:425]: `qtarr = vmap(self.mdb.qr_interp, (0, None))(Tarr, self.Tref_broadening)`

File: src/exojax/spec/opacalc.py [hunk 4]
```diff
@@ -420,9 +420,9 @@ def xsmatrix(self, Tarr, Parr):
         ) = self.opainfo
 
         if self.mdb.dbtype == "hitran":
-            qtarr = vmap(self.mdb.qr_interp, (None, 0))(self.mdb.isotope, Tarr)
+            qtarr = vmap(self.mdb.qr_interp, (None, 0, None))(self.mdb.isotope, Tarr, Tref_original)
```
Comment (HajimeKawahara) [inline src/exojax/spec/opacalc.py:423]: qtarr = vmap(self.mdb.qr_interp, (None, 0, None))(self.mdb.isotope, Tarr, self.Tref_broadening)

**Pull request merged**